### PR TITLE
Adding back files required for chef verify to work

### PR DESCRIPTION
**Pull request by tyler-ball**
Signed-off-by: tyler-ball <tball@chef.io>

### CODE REVIEW

**Comment by tyler-ball** [comment]
I still get a failure locally in the ChefDK and I'm not sure why. I copied all these files over into the current chefspec gem installed into the DK.

```
 chef verify chefspec --unit
[WARN] This is an internal command used by the ChefDK development team. If you are a ChefDK user, please do not run it.
Running verification for component 'chefspec'
.....................
No examples found.


Finished in 0.03397 seconds (files took 1.48 seconds to load)
0 examples, 0 failures




bundler: failed to load command: /opt/chefdk/embedded/bin/rake (/opt/chefdk/embedded/bin/rake)

Bundler::GitError: The git source https://github.com/chef/chef is not yet checked out. Please run `bundle install` before trying to start your application
  /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/source/git/git_proxy.rb:230:in `allowed_in_path'
  /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/source/git/git_proxy.rb:187:in `find_local_revision'
  /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/source/git/git_proxy.rb:64:in `revision'
  /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/source/git.rb:224:in `revision'
  /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/source/git.rb:92:in `install_path'
  /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/source/path.rb:126:in `expanded_path'
  /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/source/path.rb:163:in `load_spec_files'
  /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/source/git.rb:199:in `load_spec_files'
  /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/source/path.rb:100:in `local_specs'
  /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/source/git.rb:166:in `specs'
  /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/definition.rb:751:in `block in converge_locked_specs'
  /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/definition.rb:737:in `each'
  /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/definition.rb:737:in `converge_locked_specs'
  /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/definition.rb:247:in `resolve'
  /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/definition.rb:170:in `specs'
  /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/definition.rb:237:in `specs_for'
  /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/definition.rb:226:in `requested_specs'
  /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/runtime.rb:108:in `block in definition_method'
  /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/runtime.rb:20:in `setup'
  /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler.rb:107:in `setup'
  /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/setup.rb:20:in `<top (required)>'
  /opt/chefdk/embedded/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
  /opt/chefdk/embedded/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'


---------------------------------------------
Verification of component 'chefspec' failed.
```

Its failing because the chefspec Gemfile pulls chef from the git reference, but I'm not sure how this has ever been passing

File: chefspec.gemspec [hunk 1]
```diff
@@ -18,7 +18,8 @@ Gem::Specification.new do |s|
   s.license       = 'MIT'
 
   # Packaging
-  s.files         = %w{LICENSE} + Dir.glob("{lib,templates}/**/*", File::FNM_DOTMATCH).reject { |f| File.directory?(f) }
+  s.files         = %w{LICENSE Rakefile Gemfile chefspec.gemspec} +
+    Dir.glob("{lib,templates}/**/*", File::FNM_DOTMATCH).reject { |f| File.directory?(f) }
```
Comment (lamont-granquist) [inline chefspec.gemspec]: you need the 'spec' dir here as well don't you?

**Comment by lamont-granquist** [comment]
> Its failing because the chefspec Gemfile pulls chef from the git reference, but I'm not sure how this has ever been passing

so chef verify is supposed to test it against the merged gemfile.lock for chefspec.

ah, but chefspec isn't appbundled because its api is just rspec.

so yeah, IDK.

berkshelf has this lockfile:

```
-rw-r--r--  1 root  wheel  24374 Dec  6 14:11 /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/berkshelf-7.0.7/.appbundler-gemfile20181206-75307-8jd6f8.lock
```

that is supposed to be the transitive merge between the chefdk Gemfile.lock (which contains berkshelf's runtime gems) and berkshelf's development deps.

this is done to avoid the problem of having to solve for every gem's dev deps going into the mega-Gemfile.lock in DK.

chefspec seems to violate that because it isn't appbundled, so i don't know exactly how it was ever passing either...

**Comment by tas50** [comment]
Fixed in #946

**Pull request closed without merging**